sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs

```diff
@@ -202,18 +225,22 @@ public virtual async Task<Response> DeleteUserAsync(CommunicationUserIdentifier
             }
         }
 
-        /// <summary>Gets a token for a <see cref="CommunicationUserIdentifier"/>.</summary>
-        /// <param name="communicationUser">The <see cref="CommunicationUserIdentifier"/> for whom to get a token.</param>
-        /// <param name="scopes">The scopes that the token should have.</param>
+        /// <summary>Gets a Communication Identity access token for a <see cref="CommunicationUserIdentifier"/>.</summary>
+        /// <param name="communicationUser">The <see cref="CommunicationUserIdentifier"/> for whom to get a Communication Identity access token.</param>
+        /// <param name="scopes">List of <see cref="CommunicationTokenScope"/> scopes for the token.</param>
+        /// <param name="tokenExpiresAfter">Custom validity period of the Communication Identity access token within &lt;60,1440&gt; minutes range. If not provided, the default value of 1440 minutes (24 hours) will be used.</param>
         /// <param name="cancellationToken">The cancellation token to use.</param>
         /// <exception cref="RequestFailedException">The server returned an error.</exception>
-        public virtual Response<AccessToken> GetToken(CommunicationUserIdentifier communicationUser, IEnumerable<CommunicationTokenScope> scopes, CancellationToken cancellationToken = default)
+        public virtual Response<AccessToken> GetToken(CommunicationUserIdentifier communicationUser, IEnumerable<CommunicationTokenScope> scopes, TimeSpan tokenExpiresAfter, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CommunicationIdentityClient)}.{nameof(GetToken)}");
             scope.Start();
             try
             {
-                Response<CommunicationIdentityAccessToken> response = RestClient.IssueAccessToken(communicationUser.Id, scopes, cancellationToken);
+                int? expiresAfter = tokenExpiresAfter == default ? null : Convert.ToInt32(tokenExpiresAfter.TotalMinutes);
```

Comment:
I was giving this a thought. It's not a service parameter validation per se, the service will never accept anything else than int minutes for this param. However, it's possible that this value overflows, e.g. for `new TimeSpan(int.MaxValue/20, 0, 0)` and you get just a generic message saying `'Value was either too large or too small for an Int32.'`.

So, I was thinking we could catch the `System.OverflowException` and enrich the exception with a message such as `The tokenExpiresAfter argument is out of permitted bounds. Please refer to the documentation and set the value accordingly. `

Because Contoso might not know what int we're talking about - the exception doesn't contain the name of the param and they're setting `TimeSpan`, not `Int32`.

Reply:
wdyt @AikoBB ?

Reply:
yeah, totally agree, we could handle the integer overflow scenarios(for other platforms as well), good catch!

Reply:
Add exception handling. Not closing this comment. Please check if handled correctly.

Reply:
Couple of things I'd suggest:
- Scope the try-catch only to the one line that's in question. 
- Inline the message and utilize `nameof` for the param name
- Extract the logic to a method `GetTokenExpirationInMinutes()` and do the try-catch there.

Here's a great example that we can follow:
- https://github.com/Azure/azure-sdk-for-net/blob/c255d5ccff563083dac78e1c41b190ec1401a355/sdk/translation/Azure.AI.Translation.Document/src/ClientCommon.cs

Reply:
Updated.